tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml

```diff
@@ -45,38 +50,56 @@ jobs:
 
   - template: use-android-ndk.yml
 
-  - template: use-android-emulator.yml
-    parameters:
-      create: true
-      start: true
-
   - script: |
-      set -e -x
-      mkdir android_test
-      cd android_test
-      cp -av $(Build.SourcesDirectory)/java/src/test/android ./
-      cd ./android
-      mkdir -p app/libs
-      cp $(Build.BinariesDirectory)/final-android-aar/${{parameters.packageName}}-$(OnnxRuntimeVersion).aar app/libs/onnxruntime-android.aar
-      $(Build.SourcesDirectory)/java/gradlew --no-daemon clean connectedDebugAndroidTest --stacktrace
-    displayName: Run E2E test using Emulator
+        set -e -x
+        mkdir -p android_test/android/app/libs
+        cd android_test/android
+        cp -av $(Build.SourcesDirectory)/java/src/test/android/* ./
+        cp $(Build.BinariesDirectory)/final-android-aar/${{parameters.packageName}}-$(OnnxRuntimeVersion).aar app/libs/onnxruntime-android.aar
+    displayName: Copy Android test files and AAR to android_test directory
     workingDirectory: $(Build.BinariesDirectory)
 
-  - template: use-android-emulator.yml
-    parameters:
-      stop: true
+  # skip emulator tests for qnn package as there are no arm64-v8a emulators
+  - ${{ if not(contains(parameters.packageName, 'qnn')) }}:
+    - template: use-android-emulator.yml
+      parameters:
+        create: true
+        start: true
+
+    - script: |
+        set -e -x
+        cd android_test/android
+        $(Build.SourcesDirectory)/java/gradlew --no-daemon clean connectedDebugAndroidTest --stacktrace
+      displayName: Run E2E test using Emulator
+      workingDirectory: $(Build.BinariesDirectory)
+
+    - template: use-android-emulator.yml
+      parameters:
+        stop: true
 
   # we run e2e tests on one older device (Pixel 3) and one newer device (Galaxy 23)
   - script: |
       set -e -x
+      cd android_test/android
+      # build apks for qnn package as they are not built in the emulator test step
+      if [[ ${{ parameters.packageName }} == *qnn* ]]; then
+          # qnn-runtime package is expected to be of the form qnn-runtime <major>.<minor>.<patch>
+          qnn_version=$(echo "${{ parameters.QnnSDKVersion }}" | cut -d'.' -f1-3)
+          $(Build.SourcesDirectory)/java/gradlew  --no-daemon clean assembleDebug assembleAndroidTest -DqnnVersion=$qnn_version --stacktrace
+      fi
+
       pip install requests
+      cd app/build/outputs/apk
+
       python $(Build.SourcesDirectory)/tools/python/upload_and_run_browserstack_tests.py \
         --test_platform espresso \
         --app_apk_path "debug/app-debug.apk" \
         --test_apk_path "androidTest/debug/app-debug-androidTest.apk" \
-        --devices "Samsung Galaxy S23-13.0" "Google Pixel 3-9.0"
+        --devices "Samsung Galaxy S23-13.0" "Google Pixel 3-9.0" \
+        --build_tag "${{ parameters.packageName }}"
```

Comment:
Should the `--project` param be used to clarify if it's a QNN/non-QNN run?

Reply:
The project param groups the runs together- we can use if we want to separate the qnn runs

Reply:
It shows up like this
![image](https://github.com/user-attachments/assets/c4fba8ac-adca-4e88-8a14-742834eae2fe)

Reply:
As long as `onnxruntime-android-qnn` (or assumably `onnxruntime-android` for the non-QNN build) is that visible I think it's clear enough.